### ISSUE TITLE
[EventDispatcher] - broken registration of subscribers/listeners via RegisterListenersPass

### ISSUE THREAD

**Issue by K0nias** [symfony/symfony#24272]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 2.5.* - 3.2.*

Symfony\Component\EventDispatcher\DependencyInjection\RegisterListenersPass trys to reggister EventListeners and EventSubscribers using  those methods:

EventDispatcher::addListenerService
EventDispatcher::addSubscriberService

it should use those methods: 

EventDispatcher::addListener
EventDispatcher::addSubscriber

**Comment by stof** [comment]
No, for these versions of Symfony, it should not. The compiler pass in these versions is meant to configure a ContainerAwareEventDispatcher, which has these methods.
Using `addSubscriber` in these versions would skip lazy-loading (all listeners for all events would be instantiated when instantiating the dispatcher).
